src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/LayersNamesAliasingTests.java

```diff
@@ -0,0 +1,233 @@
+                "//wfs:FeatureCollection/wfs:member/st_gml32:Station_gml32");
+    }
+
+    @Test
+    public void testGetFeaturePostQuery20LocalWspace() throws Exception {
+        String xmlQuery = resourceToString("wfs200queryNoPrefix.xml");
+        Document document = postAsDOM("st_gml32/wfs", xmlQuery);
+        checkCount(
+                WFS20_XPATH_ENGINE,
+                document,
+                1,
+                "//wfs:FeatureCollection/wfs:member/st_gml32:Station_gml32");
+    }
+
+    private String resourceToString(String filename) throws IOException {
+        return IOUtils.toString(
+                getClass()
+                        .getClassLoader()
+                        .getResourceAsStream("test-data/stations/layerNamesTest/" + filename),
+                StandardCharsets.UTF_8);
+    }
+
+    @Before
+    public void beforeTest() {
+        // instantiate WFS 1.1 xpath engine
+        WFS11_XPATH_ENGINE =
+                StationsMockData.buildXpathEngine(
+                        getBaseNamespaces(),
+                        "wfs",
+                        "http://www.opengis.net/wfs",
+                        "gml",
+                        "http://www.opengis.net/gml");
+        // instantiate WFS 2.0 xpath engine
+        WFS20_XPATH_ENGINE =
+                StationsMockData.buildXpathEngine(
+                        getBaseNamespaces(),
+                        "wfs",
+                        "http://www.opengis.net/wfs/2.0",
+                        "gml",
+                        "http://www.opengis.net/gml/3.2");
+    }
+
+    protected Map<String, String> getBaseNamespaces() {
+        Map<String, String> nss = new HashMap<>();
+        nss.put("st_gml31", "http://www.stations_gml31.org/1.0");
+        nss.put("ms_gml31", "http://www.measurements_gml31.org/1.0");
+        nss.put("st_gml32", "http://www.stations_gml32.org/1.0");
+        nss.put("ms_gml32", "http://www.measurements_gml32.org/1.0");
+        return nss;
+    }
+
+    /**
+     * Helper method that checks if the provided XPath expression evaluated against the provided XML
+     * document yields the expected number of matches.
+     */
+    private void checkCount(
+            XpathEngine xpathEngine, Document document, int expectedCount, String xpath) {
+        try {
+            // evaluate the xpath and compare the number of nodes found
+            assertEquals(expectedCount, xpathEngine.getMatchingNodes(xpath, document).getLength());
+        } catch (Exception exception) {
+            throw new RuntimeException("Error evaluating xpath.", exception);
+        }
+    }
+
+    public static void copyResource(String resourcePath, Path directoryPath) throws IOException {
+        ClassLoader cl = LayersNamesAliasingTests.class.getClassLoader();
+        Files.copy(cl.getResourceAsStream(resourcePath), directoryPath);
+    }
+
+    /** * GetFeature tests ** */
+    @Override
+    protected AliasStationsMockData createTestData() {
```

Comment:
I would keep code that ``works on the setup of the tests`` together, at beginning or at the end (IMHO it should at the end).